src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -0,0 +1,960 @@
+    //https://bugs.launchpad.net/mixxx/+bug/1744816
+    html.append(QString("<a style=\"color:#0496FF;\" href=\"refresh\">%1</a>")
+                        .arg(refreshLink));
+    return html;
+}
+
+void RekordboxFeature::refreshLibraryModels() {
+}
+
+void RekordboxFeature::activate() {
+    qDebug() << "RekordboxFeature::activate()";
+
+    // Usually the maximum number of threads
+    // is > 2 depending on the CPU cores
+    // Unfortunately, within VirtualBox
+    // the maximum number of allowed threads
+    // is 1 at all times We'll need to increase
+    // the number to > 1, otherwise importing the music collection
+    // takes place when the GUI threads terminates, i.e., on
+    // Mixxx shutdown.
+    QThreadPool::globalInstance()->setMaxThreadCount(4); //Tobias decided to use 4
+    // Let a worker thread do the XML parsing
+    m_devicesFuture = QtConcurrent::run(findRekordboxDevices, this);
+    m_devicesFutureWatcher.setFuture(m_devicesFuture);
+    m_title = tr("(loading) Rekordbox");
+    //calls a slot in the sidebar model such that 'Rekordbox (isLoading)' is displayed.
+    emit(featureIsLoading(this, true));
+
+    emit(enableCoverArtDisplay(true));
+    emit(switchToView("REKORDBOXHOME"));
+}
+
+void RekordboxFeature::activateChild(const QModelIndex &index) {
+    if (!index.isValid())
+        return;
+
+    //access underlying TreeItem object
+    TreeItem *item = static_cast<TreeItem *>(index.internalPointer());
+    if (!(item && item->getData().isValid())) {
+        return;
+    }
+
+    // TreeItem list data holds 2 values in a QList and have different meanings.
+    // If the 2nd QList element IS_RECORDBOX_DEVICE, the 1st element is the
+    // filesystem device path, and the parseDeviceDB concurrent thread to parse
+    // the Rekcordbox database is initiated. If the 2nd element is
+    // IS_NOT_RECORDBOX_DEVICE, the 1st element is the playlist path and it is
+    // activated.
+    QList<QVariant> data = item->getData().toList();
+    QString playlist = data[0].toString();
+    bool doParseDeviceDB = data[1].toString() == IS_RECORDBOX_DEVICE;
+
+    qDebug() << "RekordboxFeature::activateChild " << item->getLabel()
+             << " playlist: " << playlist << " doParseDeviceDB: " << doParseDeviceDB;
+
+    if (doParseDeviceDB) {
+        qDebug() << "Parse Rekordbox Device DB: " << playlist;
+
+        // Usually the maximum number of threads
+        // is > 2 depending on the CPU cores
+        // Unfortunately, within VirtualBox
+        // the maximum number of allowed threads
+        // is 1 at all times We'll need to increase
+        // the number to > 1, otherwise importing the music collection
+        // takes place when the GUI threads terminates, i.e., on
+        // Mixxx shutdown.
+        QThreadPool::globalInstance()->setMaxThreadCount(4); //Tobias decided to use 4
+        // Let a worker thread do the XML parsing
+        m_tracksFuture = QtConcurrent::run(parseDeviceDB, m_database, item);
```

Comment:
Reusing the SQLite database connection in another thread is not permitted! The TraktorFeature does it too, but this is wrong and causes undefined behaviour.